client/core/core.go

```diff
@@ -3234,17 +3256,10 @@ out:
 			dc.tradeMtx.Lock()
 			for oid, trade := range dc.trades {
 				if !trade.isActive() {
-					log.Infof("Retiring inactive order %v", oid)
-					trade.mtx.RLock()
-					if trade.change != nil {
-						changeCoins := asset.Coins{trade.change}
-						err := trade.wallets.fromWallet.ReturnCoins(changeCoins)
-						log.Debugf("Unlocked order %v change coins: %v (wallet error = %v)", oid, changeCoins, err)
-					}
-					tradeCoins := trade.coinList()
-					err := trade.wallets.fromWallet.ReturnCoins(tradeCoins)
-					log.Debugf("Unlocked funding coins for order %v: %v (wallet error = %v)", oid, tradeCoins, err)
```

Comment:
@chappjc were these intentional for debugging purposes? It's likely/expected that the funding coins or change coin for inactive orders have been returned already. Also, there'd be little need to unlock funding coins if there's a change coin (the funding coin must have been spent to create the change coin, and thus unlocked).

I've replaced these with `trade.returnCoins()` which performs selective coin unlocking. If there are issues with coin unlocking, your approach of unlocking everything and logging the response will reveal the issues. Otherwise, it'd just be spammy. What's your take?

Reply:
That's fine.  This was just the least clever and most thorough solution.  Never intended as the best solution, just a band-aid.